sdk/appconfiguration/app-configuration/src/index.ts

```diff
@@ -47,30 +47,50 @@ const deserializationContentTypes = {
 }
 
 export interface ETagOption {
+  /**
+   * Entity tag (etag) of the object
+   */
   etag?: string;
 }
 
 export type AddConfigurationSettingConfig = Pick<ModelConfigurationSetting, Exclude<keyof ModelConfigurationSetting, "key">>;
 export type AddConfigurationSettingOptions = ModelConfigurationClientCreateOrUpdateConfigurationSettingOptionalParams;
 export type AddConfigurationSettingsResponse = ModelCreateOrUpdateConfigurationSettingResponse;
+
 export type DeleteConfigurationSettingOptions = ModelConfigurationClientDeleteConfigurationSettingOptionalParams & ETagOption;
 export type DeleteConfigurationSettingResponse = ModelDeleteConfigurationSettingResponse;
+
 export type GetConfigurationSettingOptions = ModelConfigurationClientGetConfigurationSettingOptionalParams;
 export type GetConfigurationSettingResponse = ModelGetConfigurationSettingResponse;
+
 export type ListConfigurationSettingsOptions = ModelConfigurationClientListConfigurationSettingsOptionalParams;
 export type ListConfigurationSettingsResponse = ModelListConfigurationSettingsResponse;
+
 export type ListRevisionsOptions = ModelConfigurationClientListRevisionsOptionalParams;
 export type ListRevisionsResponse = ModelListRevisionsResponse;
+
 export type SetConfigurationSettingConfig = Pick<ModelConfigurationSetting, Exclude<keyof ModelConfigurationSetting, "key">>;
 export type SetConfigurationSettingOptions = ModelConfigurationClientCreateOrUpdateConfigurationSettingOptionalParams;
 export type SetConfigurationSettingResponse = ModelCreateOrUpdateConfigurationSettingResponse;
+
 export type UpdateConfigurationSettingConfig = Pick<ModelConfigurationSetting, Exclude<keyof ModelConfigurationSetting, "key">>;
 export type UpdateConfigurationSettingOptions = ModelConfigurationClientCreateOrUpdateConfigurationSettingOptionalParams;
 export type UpdateConfigurationSettingResponse = ModelCreateOrUpdateConfigurationSettingResponse;
 
+/**
+ * Client for the Azure App Configuration service.
+ */
 export class AppConfigurationClient {
   private client: ConfigurationClient;
 
+  /**
+   * Initializes a new instance of the AppConfigurationClient class.
+   * @param connectionString Connection string needed for a client to connect to Azure.
+   *//**
+* Initializes a new instance of the AppConfigurationClient class.
+* @param uri The base URI for the Azure service
+* @param credential The credentials to use for authentication.
+*/
```

Comment:
Fro constructor overloads, you can add the jsdocs above each variation.
Look at [constructors for EventHubsClient](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/eventhub/event-hubs/src/eventHubClient.ts#L232) for an example

Reply:
Done. This syntax also works (verified by generating with typedoc) but it's pretty gross comparatively.